sporkfish/endgame_tablebase.py

```diff
@@ -110,9 +118,64 @@ def _should_query(self, board: Board, max_syzygy_size: int = 6) -> bool:
                 return False
         return True
 
+    def _categorize_dtz(self, dtz: int) -> "EndgameTablebase.DTZCategory":
+        """
+        Categorizes the DTZ (Depth to Zeroing) value into different categories.
+
+        :param dtz: The DTZ value to categorize, from our perspective.
+        :type dtz: int
+        :return: The category of the DTZ value.
+        :rtype: EndgameTablebase.DTZCategory
+        """
+        assert isinstance(
+            dtz, int
+        ), f"Expected DTZ to be of type int but got {type(dtz).__name__}."
+        if dtz < -100:
+            return EndgameTablebase.DTZCategory.CURSED_LOSS
+        elif -100 <= dtz <= -1:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_LOSS
+        elif dtz == 0:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_DRAW
+        elif 1 <= dtz <= 100:
+            return EndgameTablebase.DTZCategory.UNCONDITIONAL_WIN
+        else:
+            return EndgameTablebase.DTZCategory.CURSED_WIN
+
+    def _compare_dtz(self, dtz: int, best_dtz: int, category: DTZCategory) -> int:
+        """
+        Compares two DTZ values based on their categories.
+        Returns the best resulting DTZ value, based on the condition of the category.
+        For example, for the CURSED_LOSS category, we want to save our loss as quickly as possible.
+        CURSED_LOSS values are negative, thus we want to pick the biggest one.
+
+        :param dtz: The DTZ value to compare.
+        :type dtz: int
+        :param best_dtz: The best DTZ value so far.
+        :type best_dtz: int
+        :param category: The category of the DTZ value.
+        :type category: EndgameTablebase.DTZCategory
+        :return: The best resulting DTZ value, based on condition of the category.
+        :rtype: int
+        """
+        # We want to save our cursed loss as quickly as possible
+        if category == EndgameTablebase.DTZCategory.CURSED_LOSS:
+            return max(dtz, best_dtz)
+        # We want make the unconditional last as long as possible, in case they run out of time
+        elif category == EndgameTablebase.DTZCategory.UNCONDITIONAL_LOSS:
+            return min(dtz, best_dtz)
+        elif category == EndgameTablebase.DTZCategory.UNCONDITIONAL_DRAW:
+            return 0
+        # We want to unconditionally win as quickly as possible
+        elif category == EndgameTablebase.DTZCategory.UNCONDITIONAL_WIN:
+            return min(dtz, best_dtz)
+        # We want to extend our cursed win as much as possible, in case they run out of time
+        else:
+            return max(dtz, best_dtz)
```

Comment:
They will eventually manage  to win, since I think in the same vein every move they make will reduce the DTZ. But yes attempting to stall as long as possible by taking the move with longest DTZ

Reply:
Cursed win = they are originally winning but if we play the right moves, we are saved by the 50 move rule

Reply:
I think makes sense to me.

Reply:
We probably don't have to care about correspondence chess (since it has no time limit - we don't have enough compute to constantly run the bot).